internal/services/codesigning/trusted_signing_account_resource.go

```diff
@@ -0,0 +1,222 @@
+	"github.com/hashicorp/terraform-provider-azurerm/internal/tf/pluginsdk"
+	"github.com/hashicorp/terraform-provider-azurerm/internal/tf/validation"
+)
+
+type TrustedSigningAccountModel struct {
+	Name              string            `tfschema:"name"`
+	Location          string            `tfschema:"location"`
+	ResourceGroupName string            `tfschema:"resource_group_name"`
+	AccountUri        string            `tfschema:"account_uri"`
+	SkuName           string            `tfschema:"sku_name"`
+	Tags              map[string]string `tfschema:"tags"`
+}
+
+type TrustedSigningAccountResource struct{}
+
+var _ sdk.Resource = (*TrustedSigningAccountResource)(nil)
+
+func (m TrustedSigningAccountResource) Arguments() map[string]*pluginsdk.Schema {
+	return map[string]*pluginsdk.Schema{
+		"name": {
+			Type:     pluginsdk.TypeString,
+			Required: true,
+			ForceNew: true,
+			ValidateFunc: validation.All(
+				validation.StringLenBetween(3, 24),
+				validation.StringMatch(
+					regexp.MustCompile("^[A-Za-z][A-Za-z0-9]*(?:-[A-Za-z0-9]+)*$"),
+					"An account's name must be between 3-24 alphanumeric characters. The name must begin with a letter, end with a letter or digit, and not contain consecutive hyphens.",
+				),
+			),
+		},
+
+		"location": commonschema.Location(),
+
+		"resource_group_name": commonschema.ResourceGroupName(),
+
+		"sku_name": {
+			Type:     pluginsdk.TypeString,
+			Required: true,
+			ValidateFunc: validation.StringInSlice(
+				codesigningaccounts.PossibleValuesForSkuName(),
+				false),
+		},
+
+		"tags": commonschema.Tags(),
+	}
+}
+
+func (m TrustedSigningAccountResource) Attributes() map[string]*pluginsdk.Schema {
+	return map[string]*pluginsdk.Schema{
+		"account_uri": {
+			Type:     pluginsdk.TypeString,
+			Computed: true,
+		},
+	}
+}
+
+func (m TrustedSigningAccountResource) ModelObject() interface{} {
+	return &TrustedSigningAccountModel{}
+}
+
+func (m TrustedSigningAccountResource) ResourceType() string {
+	return "azurerm_trusted_signing_account"
+}
+
+func (m TrustedSigningAccountResource) Create() sdk.ResourceFunc {
+	return sdk.ResourceFunc{
+		Timeout: 30 * time.Minute,
+		Func: func(ctx context.Context, meta sdk.ResourceMetaData) error {
+			client := meta.Client.CodeSigning.Client.CodeSigningAccounts
+
+			var model TrustedSigningAccountModel
+			if err := meta.Decode(&model); err != nil {
+				return fmt.Errorf("decoding: %+v", err)
+			}
+
+			subscriptionID := meta.Client.Account.SubscriptionId
+			id := codesigningaccounts.NewCodeSigningAccountID(subscriptionID, model.ResourceGroupName, model.Name)
+			existing, err := client.Get(ctx, id)
+			if !response.WasNotFound(existing.HttpResponse) {
+				if err != nil {
+					return fmt.Errorf("retrieving %s: %v", id, err)
+				}
+				return meta.ResourceRequiresImport(m.ResourceType(), id)
+			}
+
+			req := codesigningaccounts.CodeSigningAccount{
+				Name:     &model.Name,
+				Location: model.Location,
```

Comment:
Forgot to normalize this one as well, we normalize when building the payload as well as before setting into state
```suggestion
				Location: location.Normalize(model.Location),
```